### PR TITLE
fix: used monaco compiler options latest

### LINKED ISSUE #865: enable js to a higher version in monaco editor
Image

### PR DESCRIPTION
**Pull request by Kevin101Zhang**
fix: Set monaco compiler options to latest

### CODE REVIEW

File: frontend/src/components/Editor/EditorComponents/Editor.tsx [hunk 1]
```diff
@@ -282,8 +282,9 @@ const Editor: React.FC = (): ReactElement => {
       `${primitives}}`,
       'file:///node_modules/@near-lake/primitives/index.d.ts',
     );
+    
     monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
-      target: monaco.languages.typescript.ScriptTarget.ES2016,
+      target: monaco.languages.typescript.ScriptTarget.Latest,
```
Comment (Kevin101Zhang) [inline frontend/src/components/Editor/EditorComponents/Editor.tsx:287]: print: monaco.languages.typescript.ScriptTarget

{
    "0": "ES3",
    "1": "ES5",
    "2": "ES2015",
    "3": "ES2016",
    "4": "ES2017",
    "5": "ES2018",
    "6": "ES2019",
    "7": "ES2020",
    "99": "Latest",
    "100": "JSON",
    "ES3": 0,
    "ES5": 1,
    "ES2015": 2,
    "ES2016": 3,
    "ES2017": 4,
    "ES2018": 5,
    "ES2019": 6,
    "ES2020": 7,
    "ESNext": 99,
    "JSON": 100,
    "Latest": 99
}

**Comment by Kevin101Zhang** [comment]
Are there any issues where setting to the latest compiler options can be harmful?

**Comment by Kevin101Zhang** [comment]
Screenshot 2024-07-23 at 4 38 12 PM
Screenshot 2024-07-23 at 4 50 08 PM

**Comment by darunrs** [comment]
So the backend [runs](https://github.com/near/queryapi/blob/main/runner/tsconfig.json) with a target of es2018 and a runtime of 2021. I'm not totally sure of the difference, even with the comments there. But in any case, that's what we use. I would say maybe we should match them. I'm not clear on what the differences between the versions are, but I would surmise there's at least some, if not significant, differences. Since we use VM2 under the hood, we're kinda forced to be stuck with specific Node/TS/etc. versions. So, perhaps we can stick to either 2018 or 2021.

**Comment by Kevin101Zhang** [comment]
Sorry this got lost in the sprint cycle from last week. I modified the monaco editor to run in 2018 to match the backend.
In this case - all the frontend code is compatible with the same features as the backend target. It should make it easier to share code if necessary.

ES2021 may introduce discrepancies with the backend.

regardless .entries got introduced in 2017 so either work.

**Review by darunrs — APPROVED** [review]

**Pull request merged**